packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -412,3 +414,115 @@ class HeroParty {
 
   String toString() => '$_heroes';
 }
+
```

Comment:
please move the hero controller to this file in a separate patch so that this is easier to review